src/lib.rs

```diff
@@ -33,12 +33,126 @@ impl<'a> Image<'a> {
+                            + bottom_right * fx * fy;
+                        if let Some(alpha_cutoff) = alpha_cutoff {
+                            if alpha > alpha_cutoff {
+                                texel_coverage += 1.0;
+                            }
+                        } else {
+                            texel_coverage += alpha
+                        }
+                    }
+                }
+                coverage += texel_coverage / (subsample_factor * subsample_factor) as f32;
+            }
+        }
+
+        coverage / ((self.width - 1) * (self.height - 1)) as f32
+    }
+
+    /// Computes the scaling factor needed for the texture's alpha such that the desired
+    /// coverage is best approximated
+    /// Ported version of implementation in https://github.com/castano/nvidia-texture-tools/.
+    pub fn find_alpha_scale_for_coverage(
+        &self,
+        desired_coverage: f32,
+        alpha_cutoff: Option<f32>,
+    ) -> f32 {
+        // This range of potential scale values is an estimate. Especially the upper bound
+        // may not be sufficient for some use-cases, depending on the downscale
+        // compared to mip 0
+        // TO-DO: Figure out if this should be exposed
+        let mut alpha_scale_range_start = 0.0;
+        let mut alpha_scale_range_end = 8.0;
+
+        let mut alpha_scale = 1.0;
+
+        // The search is done heuristically, so a tested alpha scale value
+        // does not directly map to the resulting alpha coverage. To ensure,
+        // we do not overwrite the best result, store
+        // the best result so far and use that instead of the last found scale
+        let mut best_abs_diff = f32::INFINITY;
+        let mut best_alpha_scale = alpha_scale;
+
+        // 10-step binary search for the alpha multiplier that best matches
+        // the desired alpha coverage
+        for _ in 0..10 {
+            let current_coverage = self.calculate_scaled_alpha_coverage(alpha_cutoff, alpha_scale);
+            let coverage_diff = current_coverage - desired_coverage;
+
+            if coverage_diff.abs() < best_abs_diff {
+                best_abs_diff = coverage_diff.abs();
+                best_alpha_scale = alpha_scale;
+            }
+
+            if current_coverage < desired_coverage {
+                alpha_scale_range_start = alpha_scale;
+            } else if current_coverage > desired_coverage {
+                alpha_scale_range_end = alpha_scale;
+            } else {
+                break;
+            }
+            alpha_scale = (alpha_scale_range_start + alpha_scale_range_end) / 2.0
+        }
+        best_alpha_scale
+    }
+}
+
+pub fn apply_alpha_scale(data: &mut [u8], alpha_scale: f32) {
+    for pixel in data.iter_mut().skip(3).step_by(4) {
+        *pixel = (*pixel as f32 * alpha_scale).min(255.0) as u8
+    }
+}
+
+pub enum AlphaCoverageSetting {
+    None,
+    RetainAlphaCoverage { alpha_cutoff: Option<f32> },
```

Comment:
I think we should add docs for this to explain what the differences are. In addition, we could make the default this be `AlphaCoverageSetting::None`